src/consensus/aft/raft.h

```diff
@@ -1195,16 +1194,11 @@ namespace aft
           state->commit_idx);
         return;
       }
-      else if (r.prev_idx > state->last_idx)
-      {
-        RAFT_DEBUG_FMT(
-          "Recv append entries to {} from {} but prev_idx ({}) > last_idx ({})",
-          state->node_id,
-          from,
-          r.prev_idx,
-          state->last_idx);
-        return;
-      }
+      // Redundant with check on get_term_internal() at line 1149
+      // Which captures this case in every situation, except r.prev_term == 0.
+      // That only happens if r.prev_idx == 0 however, see line 1033,
+      // in which case this assertion is also true.
+      assert(r.prev_idx <= state->last_idx);
```

Comment:
This is redundant and never taken, but in a slightly subtle way. The trade-off between removing it and keeping it to make the code more robust to potential changes is less clear to me than it was at first. I think my preference goes towards keeping the branch with the comment added. @eddyashton what do you think?

Reply:
I was going to say we have few-to-no `assert`s in the current code (non-test), but actually `raft.h` already has a few. Still, I think this that removing this branch just makes us less robust. So I'd restore the branch, and upgrade the log within it from `DEBUG` to `FAIL`, with an associated comment (this _should_ never happen, but for action-at-a-distance reasons).